src/plugins/vis_type_timeseries/server/lib/vis_data/helpers/get_splits.js

```diff
@@ -42,6 +42,7 @@ export function getSplits(resp, panel, series, meta) {
       return buckets.map((bucket) => {
         bucket.id = `${series.id}:${bucket.key}`;
         bucket.label = formatKey(bucket.key, series);
+        bucket.labelFormatted = bucket.key_as_string || '';
```

Comment:
I have made some tests with various types and it seems that you are right, `key_as_string ` is used for dates. @timroes do you have more input on this?